src/queue/mod.rs

```diff
@@ -16,18 +20,25 @@ pub trait TaskCell {
 /// which make it possible to do extreme optimizations and define complicated
 /// data struct.
 pub trait TaskQueue: Clone {
-    type Consumer;
+    /// The consumer of the task queue.
+    type Consumer: Consumer;
+    /// The task cell in the queue.
     type TaskCell: TaskCell;
 
     /// Creates a queue with a promise to only use at most `con` consumers
     /// at the same time.
-    fn with_consumers(con: usize) -> Self;
+    fn new(con: usize) -> (Self, Vec<Self::Consumer>);
 
     /// Pushes a task to the queue.
-    ///
-    /// If the queue is closed, the method should behave like no-op.
     fn push(&self, task: Self::TaskCell);
+}
+
+/// The consumer of a task queue.
+pub trait Consumer {
+    /// The task cell in the queue.
+    type TaskCell: TaskCell;
 
-    /// Closes the queue so that no more tasks can be scheduled.
```

Comment:
Why remove it?

Reply:
What's the purpose of it?

Reply:
As the comment says, close the queue so that no more tasks can be scheduled. See also crossbeam-rs/crossbeam#236.

Reply:
Should it be the thread pool who prevents the tasks from being schduled?

Reply:
How can it prevent scheduling by `Remote`?

Reply:
What I thought is to reduce an atomic load by handling it inside the queue. But if level queue can't benefit from it, I also feel OK to do it in the pool part.

Reply:
I'm curious how it reduces an atomic load? The closed flag  should be shared among all senders. Change the inner code of `crossbeam_deque::Injector::push`?

Reply:
It depends on the implement details. For the original queue implement, the desired way is to change the code of `Injector`.